### ISSUE TITLE
Show correct version when using util.Properties.versionNumberString

### ISSUE THREAD

**Issue by tgodzik** [scala/scala3#22144]
## Compiler version

3.5.2

## Minimized code

```Scala
println(util.Properties.versionNumberString)
```

## Output

```scala
2.13.15
```
or 14 don't remember, but that's not important

## Expectation

It should show 3.5.2. For that to happen two steps were suggested:
- Scala 2 library should first check for scala 3 properties
- Scala 3 should have those properties

They should be something like:
```
/library3.properties
maven.version.number=3.5.2
version.number=3.5.2
```

maven version might not be needed.

First step should be adding it here, but if we hurry we can add it to Scala 2.13.16

**Comment by hamzaremmal** [comment]
> First step should be adding it here, but if we hurry we can add it to Scala 2.13.16

Why add it to 2.13.16, can't we fully solve it in Scala 3 ?

**Comment by tgodzik** [comment]
> > First step should be adding it here, but if we hurry we can add it to Scala 2.13.16
> 
> Why add it to 2.13.16, can't we fully solve it in Scala 3 ?

Can we patch the scala library to fix that?

**Comment by sjrd** [comment]
No, we cannot override class files that exist in the scala-library.jar.

**Comment by tgodzik** [comment]
> The version of the scala-library is 2.13.15. Perhaps you want:
> 
> ```scala
> scala> println(dotty.tools.dotc.config.Properties.versionNumberString)
> 3.6.4-RC1-bin-20241204-a67dbb0-NIGHTLY
> ```

I think that requires you to have the compiler on your classpath, no?

**Comment by tgodzik** [comment]
So normal users will try to use `println(util.Properties.versionNumberString)` which will point them to thinking they are using Scala 2.13. This has actually happened, people were reporting that a wrong version is being used because that println returned 2.13. So this is quite confusing.

**Comment by hamzaremmal** [comment]
This should be fix when we work on https://github.com/scala/scala3/issues/22043

**Comment by dwijnand** [comment]
> So normal users will try to use `println(util.Properties.versionNumberString)` which will point them to thinking they are using Scala 2.13. This has actually happened, people were reporting that a wrong version is being used because that println returned 2.13. So this is quite confusing.

I take it back.  I think it's reasonable to configure scala-library's scala.util.Properties to try to look for a properties file which we'll ship in scala3-library, as an override.
